### DISCUSSION TITLE
WebDAV

### DISCUSSION THREAD

**Discussion by realchonk** [rwf2/Rocket#2524] [Questions]
Hi,

I'm currently writing a Nextcloud clone.
Is there a way to do WebDAV with Rocket?

I looked at [webdav-server-rs](https://github.com/miquels/webdav-server-rs),
but I couldn't get it to work with Rocket.

**Comment by InstinctBas** [comment]
I'm guessing webdav-server-rs has its own routing system. I've got webDav on my list to tackle at some point in the near future as well. I've been using SabreDAV for some years now, a PHP based webDav system, which has a fairly unique approach that makes it very flexible. I think rocket could be a good base for a rust implementation.

If nobody has tried doing so yet it would be interesting to see what it will take to support the extra http commands DAV needs. It's been awhile since I last dove into it but could be a fun project.

**Reply by zxq5-dev** [reply]
yeah this would be so useful for me right now. I'm genuinely so sick of how slow my self hosted nextcloud instance is and even opening my remote folder doesn't work 50% of the time
